x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/JdbcDateUtils.java

```diff
@@ -41,9 +42,25 @@
         .appendFraction(MILLI_OF_SECOND, 3, 3, true)
         .appendOffsetId()
         .toFormatter(Locale.ROOT);
+
+    static final DateTimeFormatter ISO_WITH_MINUTES = new DateTimeFormatterBuilder()
+        .parseCaseInsensitive()
+        .append(ISO_LOCAL_DATE)
+        .appendLiteral('T')
+        .appendValue(HOUR_OF_DAY, 2)
+        .appendLiteral(':')
+        .appendValue(MINUTE_OF_HOUR, 2)
+        .appendOffsetId()
+        .toFormatter(Locale.ROOT);
     
     static long asMillisSinceEpoch(String date) {
-        ZonedDateTime zdt = ISO_WITH_MILLIS.parse(date, ZonedDateTime::from);
+        ZonedDateTime zdt;
+        try {
+            zdt = ISO_WITH_MILLIS.parse(date, ZonedDateTime::from);
+        } catch (DateTimeParseException e) {
+            // Case of a Group By with casting as DATE
```

Comment:
Instead of handling the exception, it's better to modify the formatter instead to make some parsing optional (`optionalStart()`, `optionalEnd()`).
I'm unclear though why would a GROUP BY cast on a DATE return an ISO date without seconds...

Reply:
I was wrong. Forgot to revisit this part. Check: https://github.com/elastic/elasticsearch/pull/37693/commits/7409ce46df663bd65bc0662c6ae0dc42a39acac8